File: assignment 6/c/programs/8.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_SIZE 100

int stack[MAX_SIZE];
int top = -1;

void push(int element)
{
    if (top == MAX_SIZE - 1)
    {
        printf("Stack overflow\n");
        return;
    }
    stack[++top] = element;
}

int pop()
{
    if (top == -1)
    {
        printf("Stack underflow\n");
        return -1;
    }
    return stack[top--];
}

void decimalToBinary(int decimalNumber)
{
    while (decimalNumber != 0)
    {
        int remainder = decimalNumber % 2;
        push(remainder);
        decimalNumber /= 2;
    }
    while (top != -1)
    {
        printf("%d", pop());
    }
}

int main()
{
    int decimalNumber;
    printf("Enter a decimal number: ");
    scanf("%d", &decimalNumber);
    printf("The binary equivalent is: ");
    decimalToBinary(decimalNumber);
    printf("\n");
    return 0;
}

File: assignment 6/python/programs/8.py
class Stack:
    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        return self.items.pop()

    def is_empty(self):
        return len(self.items) == 0


def decimal_to_binary(decimal_num):
    stack = Stack()
    while decimal_num > 0:
        remainder = decimal_num % 2
        stack.push(remainder)
        decimal_num //= 2

    binary_str = ""
    while not stack.is_empty():
        binary_str += str(stack.pop())

    return binary_str


# Example usage
decimal_num = int(input("Enter a decimal number: "))
binary_num = decimal_to_binary(decimal_num)
print(f"The binary representation of {decimal_num} is {binary_num}.")
